binder/Dockerfile

```diff
@@ -0,0 +1,26 @@
+FROM python:3.9-slim
+# install the notebook package
+RUN pip install --no-cache --upgrade pip && \
+    pip install --no-cache notebook jupyterlab 
+
+# create user with a home directory
+ARG NB_USER
+ARG NB_UID
+ENV USER ${NB_USER}
+ENV HOME /home/${NB_USER}
+
+RUN adduser --disabled-password \
+    --gecos "Default user" \
+    --uid ${NB_UID} \
+    ${NB_USER}
+WORKDIR ${HOME}
+USER ${USER}
+
+# Make sure the contents of our repo are in ${HOME}
+COPY . ${HOME}
```

Comment:
The `binder` folder is special for binder in a way that it can contain the binder specific config files independent of the repository. This Dockerfile is then ran from the base of the directory if my understanding is correct.